src/mono/mono/metadata/class-init.c

```diff
@@ -735,6 +735,17 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 		}
 	}
 
+	// compute is_exception_class, used by interp to avoid inlining exception handling code
+	if (
+		klass->parent && !m_class_is_valuetype (klass) &&
+		!m_class_is_interface (klass)
+	) {
+		if (m_class_is_exception_class (klass->parent))
+			klass->is_exception_class = 1;
```

Comment:
you also need something in`mono_class_create_generic_inst` to copy the bit from `gklass` to `klass`